daphne/src/hpke.rs

```diff
@@ -120,30 +176,74 @@ impl HpkeReceiverConfig {
         enc: &[u8],
         ciphertext: &[u8],
     ) -> Result<Vec<u8>, HpkeError> {
-        let sk = <kem::X25519HkdfSha256 as Kem>::PrivateKey::from_bytes(&self.secret_key)?;
-        let enc = <kem::X25519HkdfSha256 as Kem>::EncappedKey::from_bytes(enc)?;
-        let mut receiver = hpke::setup_receiver::<
-            aead::AesGcm128,
-            kdf::HkdfSha256,
-            kem::X25519HkdfSha256,
-        >(&OpModeR::Base, &sk, &enc, info)?;
-        receiver.open(ciphertext, aad)
+        self.check_suite().map_err(|_| HpkeError::OpenError)?; // Note(caw): any error will do here
```

Comment:
We could change the error type from `HpkeError` to `DapError`, then the `?` should work without mapping the error first.

Reply:
Done.